### DISCUSSION TITLE
How to use redirect, it is not working

### DISCUSSION THREAD

**Discussion by wildfrontend** [remix-run/react-router#9898] [Q&A]
```
 <Button onClick={() =>   redirect(link)}/>
```

button is cliecked , and not working, and i try `useNavigate` is work
i use `createBrowserRouter` and `RouterProvider 
package.json
```
 "react-dom": "^18.2.0",
 "react-router-dom": "^6.6.1",
```

**Comment by machour** [comment]
`redirect()` creates a Response object meant to be returned from either your `loader` or `action`.

In your case, you can use navigate, or better, use a `<Link to={link}>Click me</Link>`

**Reply by wildfrontend** [reply]
oh i see, in document meant , when use `loader` or `action`, `redirect` is better than `navgiate` ,right?

**Reply by machour** [reply]
Yes. You can't use navigate in loader or action, you use redirect() there or anything else that generates a Response.

**Comment by mattes3** [comment]
I tried to use `redirect()` in loader but it doesn't work.

This is what I tried:

```jsx
<Route path="/" loader={() => redirect('/dashboard')} />
```

This is what happened in the console:

```
Matched leaf route at location "/" does not
have an element. This means it will render
an <Outlet /> with a null value by default 
resulting in an "empty" page.
```

When I write this, the console warning disappears but the page is still empty:

```jsx
<Route path="/" loader={() => redirect('/dashboard')} element={<div />} />
```

How do I write a correct redirect in a Route?
